client/api/go-client/client.go

```diff
@@ -189,11 +193,12 @@ func (c *Client) doBasic(req *http.Request) (*http.Response, error) {
 		<-c.throttle
 	}()
 
-	httpClient := &http.Client{}
-	if c.tlsClientConfig != nil {
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: c.tlsClientConfig,
-		}
+	httpClient := &http.Client{
+		CheckRedirect: c.checkRedirect,
```

Comment:
Looks like this is redundant with `httpClient.CheckRedirect = c.checkRedirect` below.

Reply:
Oops, good catch.